### PR TITLE
doc: add security section to debugging guides

### PR DESCRIPTION
**Pull request by ofrobots**
This talks about the security implications a bit more, in light of https://nodejs.org/en/blog/vulnerability/march-2018-security-releases/#node-js-inspector-dns-rebinding-vulnerability-cve-2018-7160

### CODE REVIEW

**Review by marswong — COMMENTED** [review]
good job. but was this too long for a "getting started" guide of "debugging"? what about creating a new doc for security and link here?

**Review by daxlab — COMMENTED** [review]
👍

**Review by ckotzbauer — APPROVED** [review]

File: locale/en/docs/guides/debugging-getting-started.md [hunk 1]
```diff
@@ -47,6 +47,47 @@ OS X). As of Node 7 this activates the legacy Debugger API; in Node 8 and later
...
+### Local applications have full access to the inspector
+
+Even if you bind the inspector port to 127.0.0.1 (the default), any applications
+running locally on your machine will have unrestricted access. This is by design
+to allow local debuggers to be able to attach conveniently.
+
+### Browsers, WebSockets and same-origin policy
+
+Websites open in a web-browser can make WebSocket and HTTP requests under the
+browser security model. A initial HTTP connection is necessary to obtain a
```
Comment (vsemozhetbyt) [inline locale/en/docs/guides/debugging-getting-started.md]: Nit: `A initial` -> `An initial`.

File: locale/en/docs/guides/debugging-getting-started.md [hunk 2]
```diff
@@ -161,6 +202,36 @@ The following table lists the impact of various runtime flags on debugging:
...
+
+Let's say you are running Node on remote machine, remote.example.com, that you
+want to be able to debug. On that machine, you should start the node process
+with the inspector listening only to localhost (the default).
+
+```sh
+$ node --inspect server.js
+```
+
+Now, on your local machine from where you want initiate a debug client
```
Comment (vsemozhetbyt) [inline locale/en/docs/guides/debugging-getting-started.md]: Nit: `want initiate` -> `want to initiate`.

File: locale/en/docs/guides/debugging-getting-started.md [hunk 3]
```diff
@@ -47,6 +47,47 @@ OS X). As of Node 7 this activates the legacy Debugger API; in Node 8 and later
...
+can reach your IP address will be able to connect to the debugger without any
+restriction and will be able to run arbitrary code.
+
+By default `node --inspect` binds to 127.0.0.1. You explicitly need to provide a
+public IP address or 0.0.0.0, etc., if you intend to allow external connections
+to the debugger. Doing so may expose you a potentially significant security
+threat. We suggest you ensure appropriate firewalls and access controls in place
+to prevent a security exposure.
+
+See the section on 'Enabling remote debugging scenarios' on some advice on how
```
Comment (vsemozhetbyt) [inline locale/en/docs/guides/debugging-getting-started.md]: Nit: maybe it is worth to add a link to `#enabling-remote-debugging-scenarios`?

**Review by vsemozhetbyt — APPROVED** [review]

**Comment by ofrobots** [comment]
I think a discussion of the security risks needs to be up-front in the debugging guide. It might be possible have a summary section and link to a detailed document elsewhere. Let me take a look at refactoring.

**Comment by ofrobots** [comment]
@marswong I took a deeper look at refactoring this PR into two document, and I don't think there is a refactor into two documents that would be satisfactory:

* We need to make people aware that there are security implications. These should be in the primary debugging guide.
* The details about how to do remote debugging must also be in the primary debugging guide. Those are instructions.
* The only thing that can be split out is the nature of the security issues. I tried extracting these into a standalone doc, but I wasn't satisfied with the result because the security doc has to refer back to the guide on how to enable remote debugging in light of the security issues.

In summary, I think we should keep this a single doc. Even though this doc is called the 'Debugging Getting Started' guide, in reality it is actually _the_ 'Debugging Guide'. I think security implications should stay in this doc.

I would like to land this as-is. WDYT?

**Comment by fhemberger** [comment]
@ofrobots Can you please fix the little nits @vsemozhetbyt mentioned, and I think we are good to go. If we feel the need to restructure the document, we can do so at a later point.

**Review by lpinca — APPROVED** [review]

**Review by daxlab — APPROVED** [review]

**Comment by fhemberger** [comment]
Thank you!

**Comment by marswong** [comment]
@ofrobots good job :)
